### ISSUE TITLE
Incorrect results with a subquery in VALUES list

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#8621]
### Greenplum version or build
Tested on master, probably all versions.

### Expected behavior

This is variation of a test query in the regression suite:

```
create table foo (i int4);
insert into foo values (1);

postgres=# select val.x
  from generate_series(1,10) as s(i),
  lateral (
    values ((select s.i + 1 from foo)), (s.i + 101)
  ) as val(x)
where s.i < 10 and (select val.x) < 110;
  x  
-----
   2
 102
   3
 103
   4
 104
   5
 105
   6
 106
   7
 107
   8
 108
   9
 109
  10
(17 rows)
```

### Actual behavior
```
postgres=# select val.x
  from generate_series(1,10) as s(i),
  lateral (
    values ((select s.i + 1 from foo)), (s.i + 101)
  ) as val(x)
where s.i < 10 and (select val.x) < 110;
  x  
-----
 102
 103
 104
 105
 106
 107
 108
 109
(8 rows)
```

This involves LATERAL, so I'm not sure how much we care about that. But it works if you take the 'foo' table out of the picture:
```
postgres=# select val.x
  from generate_series(1,10) as s(i),
  lateral (
    values ((select s.i + 1 /* from foo */ )), (s.i + 101)
  ) as val(x)
where s.i < 10 and (select val.x) < 110;
  x  
-----
   2
 102
   3
 103
   4
 104
   5
 105
   6
 106
   7
 107
   8
 108
   9
 109
  10
(17 rows)
```

The cause is pretty simple: we're missing handling of the VALUES lists in `plan_tree_mutator()` function:
```
diff --git a/src/backend/cdb/cdbplan.c b/src/backend/cdb/cdbplan.c
index c3118725999..bd187b6f2bd 100644
--- a/src/backend/cdb/cdbplan.c
+++ b/src/backend/cdb/cdbplan.c
@@ -516,6 +516,7 @@ plan_tree_mutator(Node *node,
 				ValuesScan *newscan;
 
 				FLATCOPY(newscan, scan, ValuesScan);
+				MUTATE(newscan->values_lists, scan->values_lists, List *);
 				SCANMUTATE(newscan, scan);
 				return (Node *) newscan;
 			}
```

**Comment by hlinnaka** [comment]
We should check if we're missing any more Plan fields in both `plan_tree_walker()` and `plan_tree_mutator()`.

**Comment by hlinnaka** [comment]
Fixed by commit 88c9977317.
